syntax = "proto3";

package auth;

// пользователь 
message User {
    int64 id = 1;
    string login = 2;
    string email = 3;
    string first_name = 4;
    string last_name = 5;
    string birth_date = 6; 
    string phone = 7;
    string created_at = 8;
    string updated_at = 9;
}

// регистрация
message RegisterUserRequest {
    string login = 1;
    string password = 2;
    string email = 3;
}

// аутентификация
message LoginRequest {
    string login = 1;
    string password = 2;
}
// ответ на аутентификацию
message LoginResponse {
    string token = 1;
    User user = 2;
}

// обновление профиля
message UpdateProfileRequest {
    string token = 1;
    string first_name = 2;
    string last_name = 3;
    string birth_date = 4;
    string email = 5;
    string phone = 6;
}

// получение профиля
message GetProfileRequest {
    string token = 1;
}

// определение сервиса аутентификации
service AuthService {
    rpc RegisterUser (RegisterUserRequest) returns (User);
    rpc Login (LoginRequest) returns (LoginResponse);
    rpc UpdateProfile (UpdateProfileRequest) returns (User);
    rpc GetProfile (GetProfileRequest) returns (User);
}
